kernel/sched.c

```diff
@@ -853,7 +853,7 @@ void _impl_k_wakeup(k_tid_t thread)
 		return;
 	}
 
-	if (_abort_thread_timeout(thread) == _INACTIVE) {
+	if (_abort_thread_timeout(thread)) {
```

Comment:
Nitpick on behalf of @ceolin : MISRA rules disallow implicit conversion to bool

But I think this is deeper.  The old code would store the "was linked" state implicit in the timeout, but that's no longer ture, right?  You don't know that a zero return here means "inactive", I think?   Might need to be reworked to do a is_linked call first and then use that as the predicate for the if()?

Actually a quick grep shows that this one of two places actually using the return value of _abort_(thread_?)timeout().  Maybe the simpler thing to do would be to just remove that return value and/or swap it to a boolean that indicates "was active".

Reply:
The old code returned zero on successful abort, and `_INACTIVE` on failure.  Returning `-EINVAL` is more consistent with indicating that the abort failed.  I really think this is doing the right thing.

I'll change the test to detect a negative result, though, instead of implicit bool although there's an awful lot of implicit bool in Zephyr.